ciphers/base32.py

```diff
@@ -1,42 +1,49 @@
-import base64
+"""
+Base32 encoding and decoding
 
+https://en.wikipedia.org/wiki/Base32
+"""
+B32_CHARSET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ234567"
 
-def base32_encode(string: str) -> bytes:
+
+def base32_encode(data: bytes) -> bytes:
     """
-    Encodes a given string to base32, returning a bytes-like object
-    >>> base32_encode("Hello World!")
+    >>> base32_encode(b"Hello World!")
     b'JBSWY3DPEBLW64TMMQQQ===='
-    >>> base32_encode("123456")
+    >>> base32_encode(b"123456")
     b'GEZDGNBVGY======'
-    >>> base32_encode("some long complex string")
+    >>> base32_encode(b"some long complex string")
     b'ONXW2ZJANRXW4ZZAMNXW24DMMV4CA43UOJUW4ZY='
     """
-
-    # encoded the input (we need a bytes like object)
-    # then, b32encoded the bytes-like object
-    return base64.b32encode(string.encode("utf-8"))
+    binary_data = "".join([bin(ord(d))[2:].zfill(8) for d in data.decode("utf-8")])
+    binary_data = binary_data.ljust(5 * ((len(binary_data) // 5) + 1), "0")
+    b32_chunks = map("".join, zip(*[iter(binary_data)] * 5))
+    b32_result = "".join(
+        j for j in [B32_CHARSET[int(chunk, 2)] for chunk in b32_chunks]
+    )
+    return bytes(b32_result.ljust(8 * ((len(b32_result) // 8) + 1), "="), "utf-8")
 
 
-def base32_decode(encoded_bytes: bytes) -> str:
+def base32_decode(data: bytes) -> bytes:
     """
-    Decodes a given bytes-like object to a string, returning a string
     >>> base32_decode(b'JBSWY3DPEBLW64TMMQQQ====')
-    'Hello World!'
+    b'Hello World!'
     >>> base32_decode(b'GEZDGNBVGY======')
-    '123456'
+    b'123456'
     >>> base32_decode(b'ONXW2ZJANRXW4ZZAMNXW24DMMV4CA43UOJUW4ZY=')
-    'some long complex string'
+    b'some long complex string'
     """
-
-    # decode the bytes from base32
-    # then, decode the bytes-like object to return as a string
-    return base64.b32decode(encoded_bytes).decode("utf-8")
+    binary_chunks = "".join(
+        [
+            bin(B32_CHARSET.index(_d))[2:].zfill(5)
+            for _d in data.decode("utf-8").strip("=")
+        ]
+    )
```

Comment:
Can we directly use a generator rather than a list?
```suggestion
    binary_chunks = "".join(
            bin(B32_CHARSET.index(_d))[2:].zfill(5)
            for _d in data.decode("utf-8").strip("=")
    )
```